frontend/app/mappers/personal-information-service-mappers.server.ts

```diff
@@ -173,3 +181,32 @@ function toAddress(addressDto: AddressDto | undefined, category: string) {
     street: addressDto?.streetName,
   });
 }
+
+function toInsurancePlanApi(
+  federalDentalPlanId: string | undefined,
+  provincialTerritorialDentalPlanId: string | undefined,
+  privateDentalPlanId: string | undefined,
+): {
+  InsurancePlanIdentification?: {
+    IdentificationID: string;
+    IdentificationCategoryText?: string;
+  };
+}[] {
+  const listOfInsurancePlans = [];
+
+  if (federalDentalPlanId) {
+    listOfInsurancePlans.push({ InsurancePlanIdentification: { IdentificationID: federalDentalPlanId, IdentificationCategoryText: 'Federal' } });
+  }
+  if (provincialTerritorialDentalPlanId) {
+    listOfInsurancePlans.push({ InsurancePlanIdentification: { IdentificationID: provincialTerritorialDentalPlanId, IdentificationCategoryText: 'Provincial and Territorial' } });
+  }
+  if (privateDentalPlanId) {
+    listOfInsurancePlans.push({ InsurancePlanIdentification: { IdentificationID: privateDentalPlanId, IdentificationCategoryText: 'Private' } });
+  }
+
+  return listOfInsurancePlans;
+}
+
+function toFederalDentalCoverageIndicator(federalDentalPlanId: string | undefined): { ReferenceDataID?: string | undefined; ReferenceDataName?: string | undefined } | undefined {
```

Comment:
The `?` identifier is redundant for `ReferenceDataID` and `ReferenceDataName` as these arguments can be of the `undefined` type:

```suggestion
function toFederalDentalCoverageIndicator(federalDentalPlanId: string | undefined): { ReferenceDataID: string | undefined; ReferenceDataName: string | undefined } | undefined {
```

Reply:
Removed: 79630faaf42b451114845addf54a78e1326280e1